src/support/modify.c

```diff
@@ -523,3 +523,79 @@ __wt_modify_vector_free(WT_MODIFY_VECTOR *modifies)
         __wt_free(modifies->session, modifies->listp);
     __wt_modify_vector_init(modifies->session, modifies);
 }
+
+/*
+ * __wt_modify_reconstruct_from_upd_list --
+ *     Takes an in-memory modify and populates an update view with the reconstructed full value.
+ */
+int
+__wt_modify_reconstruct_from_upd_list(
+  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT_UPDATE_VIEW *upd_view)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_ITEM buf;
```

Comment:
Why we can't operate on upd_view's buf directly here? It seems buf is redundant to me.

Reply:
Resolved as part of c002170.